spotifycli/spotifycli.py

```diff
@@ -212,45 +246,63 @@ def show_art_url():
 
 
 def get_spotify_property(spotify_property):
-    try:
-        dbus_addr = DBusAddress(
-            bus_name="org.freedesktop.DBus",
-            object_path="/org/freedesktop/DBus",
-            interface="org.freedesktop.DBus",
+    dbus_addr = DBusAddress(
+        bus_name="org.freedesktop.DBus",
+        object_path="/org/freedesktop/DBus",
+        interface="org.freedesktop.DBus",
+    )
+    connection = open_dbus_connection(bus="SESSION")
+
+    list_names_call = new_method_call(
+        remote_obj=dbus_addr, method="ListNames", signature=""
+    )
+    reply = connection.send_and_get_reply(list_names_call)
+    if reply.header.message_type.name == 'error':
+        raise SpotifyCLIException(
+            "Could not retrieve list of services (bus names) from dbus." +
+            f"Full error text:\n{reply.header.message}"
         )
-        connection = open_dbus_connection(bus="SESSION")
-
-        list_names_call = new_method_call(
-            remote_obj=dbus_addr, method="ListNames", signature=""
-        )
-        reply = connection.send_and_get_reply(list_names_call)
-        names = reply.body[0]
-
-        client_name = f"org.mpris.MediaPlayer2.{client}"
-        mpris_name = next((name for name in names if name.startswith(client_name)), None)
-        if mpris_name is None:
-            sys.stderr.write(f"No mpris clients found for client {client}\n")
-            sys.exit(1)
-
-        spotify_dbus_addr = DBusAddress(
-            bus_name=mpris_name,
-            object_path="/org/mpris/MediaPlayer2",
-            interface="org.freedesktop.DBus.Properties"
-        )
-        get_property_call = new_method_call(
-            remote_obj=spotify_dbus_addr,
-            method="Get",
-            signature="ss",
-            body=("org.mpris.MediaPlayer2.Player", spotify_property)
-        )
-
-        reply = connection.send_and_get_reply(get_property_call)
-        body = reply.body[0]
-        return body[1]
+    names = reply.body[0]
+
+    client_name = f"org.mpris.MediaPlayer2.{client}"
+    mpris_name = next(
+        (name for name in names if name.startswith(client_name)), None
+    )
+    if mpris_name is None:
+        raise SpotifyCLIException(
+            f"No mpris clients found for {client}, is {client} running?\n")
```

Comment:
And another "known" connection -- this is where we actually check if Spotify is running or not.